File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    ranking = [i+1 for i in range(len(players))]
    a = dict(zip(ranking,players))
    b = dict(zip(players, ranking))
    
    for calling in callings:
        cur_rank = b[calling]
        pre_rank = cur_rank - 1 
        
        origin = a[pre_rank]
        a[pre_rank] = calling
        a[cur_rank] = origin
        
        b[origin] = cur_rank
        b[calling] = pre_rank
        
        
    answer = []
    
    for _,b in a.items():
        answer.append(b)
        
    return answer

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<string> solution(vector<string> players, vector<string> callings) {
    vector<string> answer;
    unordered_map<string,int> rank;
    
    int n = players.size();
    for(int i=0; i<n; i++) {
        rank[players[i]] = i;
    }
    for(auto& c : callings) {
        int i = rank[c];
        int j = i-1;
        string front = players[j];
        // 딕셔너리 순서 바꾸기
        rank[c] = j;
        rank[front] = i;
        // 배열 순서 바꾸기
        players[j] = c;
        players[i] = front;
    }
    answer = players;
    return answer;
}